File: 0204-count-primes/0204-count-primes.java
class Solution {
    public int countPrimes(int n) {
        if(n==0 || n==1)
        {
            return 0;
        }
        boolean prime[] = new boolean[n+1];
        for(int ctr=2;ctr<=n;ctr++)
        {
            prime[ctr]=true;
        }
        for(int ctr=4;ctr<=n;ctr+=2)
        {
            prime[ctr]=false;
        }
        for(int ctr=3;ctr*ctr<=n;ctr++)
        {
            if(!prime[ctr])
            {
                continue;
            }
            for(int val=2*ctr;val<=n;val+=ctr)
            {
                prime[val]=false;
            }
        }
        int primes=0;
        for(int num=2;num<n;num++)
        {
            if(prime[num])
            {
                primes++;
            }
        }
        return primes;
    }
}

File: 0204-count-primes/0204-count-primes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        if n<2:
            return 0
        arr=[0]*n
        for i in range(2,int(n**0.5)+1):
            if arr[i]==0:
                for j in range(i*i,n,i):
                    arr[j]=1
        return sum(1 for i in range(2,n) if arr[i]==0)